src/jvm/test/org/jetbrains/kotlinx/lincheck_test/representation/InterleavingAnalysisPresentInSpinCycleFirstIterationTest.kt

```diff
@@ -25,6 +26,7 @@ import java.util.concurrent.atomic.AtomicInteger
  * Checks that after we found a spin-cycle, then we will consider interleavings with switches
  * inside the first spin-cycle iteration.
  */
+@IgnoreInTraceDebuggerMode
```

Comment:
Can we not ignore but have a different output file when `inTraceDebuggerMode == true`?

Reply:
Most of the tests, failing because of LoopDetector disabling, fail with `StackOverflow` and other similar non-recoverable errors.

Reply:
Lincheck's loop detector tries to understand the loop bounds and, thus, minimize both the execution and the output. In the trace debugger mode, this functionality is switched off. Instead, a simpler strategy is used -- when N events have occurred, Lincheck tries to switch to another thread. Besides, when the total number of events exceeds some limits, Lincheck reports a livelock. This means the loop detection should still work but the output is expected to be more verbose.

Reply:
> This means the loop detection should still work but the output is expected to be more verbose.

If some tests are failing (unexpectedly) with `StackOverflow` when loop detector is disabled, may be it is worth investigating under separate issue/PR?